### ISSUE TITLE
Explore widening Accessibility Supported Requirements 2 stacks on different OS

### ISSUE THREAD

**Issue by DavidMacDonald** [w3c/wcag21#298]
There was a thread which sprung up on Twitter regarding WCAG conformance requirements,  @stevefaulkner said that on the Open Web we should require sites to work on both Windows or Mac platforms. Others commented similarly.

https://twitter.com/davidmacd/status/870648479886053376

WCAG presently requires at least one accessibility supported way to meet each Success Criterion, and they can't switch back and forth... in other words all SCs have to be met on the same stack.

An author could rely on MacOS with VoiceOver, etc to claim conformance, but in reality I've never seen that. However, its becoming more realistic. This proposal would require at least one accessibility supported way to meet all SCs on at least two OS. 

## PROS
* It would effectively put MacOS in the requirements.
* It would give people with disabilities more choices
* Would motivate stakeholders in the Mac world to take accessibility more seriously.
* [feel free to add more in the comments, I'll add them here]

## CONS

* would increase QA and testing significantly
* Increase workload on developers to provide work around solutions for two separate stacks. At least on Windows if JAWS doesn't work on something NVDA might. On MACOS there's only VoiceOver, so when there is a bug, there is no accessibility supproted way of meeting the SC, which would...
* put sites at risk of non-conformance if there are bugs in one or the other systems. So it would be a significant expansion of the requirements of WCAG.
* Windows is still ahead of MAC on most metrics for accessibility, and most blind individuals who are serious about working and employment still rely on Windows. If they have a Mac, its usually their second machine. (conversely for Mobile, iOS is the clear leader)
* could inhibit uptake of WCAG 2.1 for the CON reasons above.
* [feel free to add more in the comments, I'll add them here]

The discussion was passionate enough that I think this needs to be explored seriously by the group and a formal decision process be documented for WCAG 2.1 rather than relying on the WCAG 2.0 rationale which was that a the time Windows was really the only OS that could reliably provide a way to meet the Success Criteria.

**Comment by patrickhlauke** [comment]
skimming over https://www.w3.org/TR/WCAG20/#conformance and https://www.w3.org/TR/UNDERSTANDING-WCAG20/conformance.html it's not immediately evident to me where "WCAG presently requires at least one accessibility supported way to meet each Success Criterion" (other than by omission, not explicitly stating that things should work in more than one UA).

The understanding document does contain non-normative guidance that

> Content posted to the public Web may need to work with a broader range of user agents and assistive technologies, including older versions.

There's of course a slippery slope issue: user agents/ATs on certain platforms are quite limited or buggy - but normatively mandating that authors need to code workaround solutions to be able to pass WCAG shifts the onus purely on the developers, when really user agents/ATs should get their house in order under UAAG requirements. this makes developers responsible for browser bugs, screen reader bugs, etc 

Taking it to extreme, if an author follows correct web standards, and due to a bug on a particular platform/AT it's not working there, would the developer/developer's company be liable in a court case for not having tested that particular platform/AT and not having coded around the bug?

I'd be happy to see non-normative guidance being strengthened, but think a normative change would shift the blame/responsibility too far.

**Comment by DavidMacDonald** [comment]
@patrickhlauke 
> it's not immediately evident to me where "WCAG presently requires at least one accessibility supported way to meet each Success Criterion"

It's determined by Item 1, and Item 4 of conformance requirements.
# 1
Conformance Level: One of the following levels of conformance is met in full.
... the Web page satisfies all the Level A, Level AA and Level AAA Success Criteria, or a Level AAA conforming alternate version is provided.  https://www.w3.org/TR/WCAG20/#conformance

# 4 
Only accessibility-supported ways of using technologies are relied upon to satisfy the success criteria. 

> I'd be happy to see non-normative guidance being strengthened, but think a normative change would shift the blame/responsibility too far.

Then we totally agree ...

**Comment by steverep** [comment]
@DavidMacDonald, We can't just talk about Mac and Windows here.  Besides other mobile platforms, we need to start taking Linux more seriously when talking about accessibility support in W3C documents.  Linux accessibility has grown quite a bit and more distros are taking it seriously, and the Orca screen reader has a very active community.  It is well known that people with disabilities are very underrepresented in STEM, and Linux is used widely in STEM.

@patrickhlauke, I completely agree that normative language to put all QA on the web developer is neither fair nor practical.  However, the lines between website, UA, and AT bug or required feature are not black and white, and even documented standards can't even make them less gray in cases.  Trying to draw lines in permanent ink will only have end-users suffering in the end.

Certainly for web content guidelines, isn't the appropriate approach here to list out what content developers are doing that results in a lack of support across platforms, and turn them into success criteria?  Even if that results in a discussion of who is to blame for the lack of support, that's probably better than trying to shift it all to QA testing.

**Comment by patrickhlauke** [comment]
> isn't the appropriate approach here to list out what content developers are doing that results in a lack of support across platforms

who will be tasked with constantly keeping this list up-to-date, in line with bug fixes/changes to user agents / ATs?

**Comment by michael-n-cooper** [comment]
This is out of scope for WCAG 2.1, but we will look at this sort of issue again in Silver.
